### ISSUE TITLE
Nickname completion UI snag: add a space on case of single match

### ISSUE THREAD

**Issue by jtroup** [matterhorn-chat/matterhorn#591]
If I type a partial nick, but one close enough to guarantee a single match, and press tab, it completes the nick but does not add a space at the end.  It would be more intuitive IMO (and match the behaviour of other TUI clients such as irssi) (and not break my muscle memory 🤡) if the space were added in the case of a single match.

**Comment by jtdaugherty** [comment]
Thanks for the report!

I've just pushed some patches to `develop` to address this. Here's a summary:

* A space is now introduced immediately following *all* completions, not just a completion that arises from a single match. (I introduced a space in all cases to give consistent behavior rather than making users develop muscle memory about when a space will be introduced or won't.)
* Tab/Shift-Tab will still cycle through the matches even in the presence of the space.
* A new configuration setting, `smartediting`, was added. Its default value is `True`. When it is `True` it enables this behavior:
* Whenever a completion is filled in with Tab/Shift-Tab, if the next input character that the user enters is a punctuation character that typically comes *after* a completion, such as period, comma, etc., the punctuation character entered will *replace* the space introduced by the first bullet above. This allows users to keep their messages tidy rather than having to hit Backspace in those (many) cases where a space separating the username (channel name, etc.) is not desirable. Users not wanting this behavior can set `smartediting` to `False` in their configuration file. The set of characters that will automatically replace a trailing space in this way (following a completion) are the characters in the string `".,'\";:)]!?"`.

Ordinarily I close tickets when we push fixes to `develop`, but I'll wait for your response to this first in this case.
